File: 167. Two Sum II - Input Array Is Sorted/167.java
// 167. Two Sum II - Input Array Is Sorted solution in Java
// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/description/

class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int low = 0;
        int high = numbers.length - 1;
        int[] return_array = {-1, -1};

        while (low < high) {
            int sum = numbers[low] + numbers[high];

            if (sum == target) {
                return_array[0] = low + 1;
                return_array[1] = high + 1;

                return return_array;
            } else if (sum < target) {
                low++;
            } else {
                high--;
            }
        }

        return return_array;
    }
}

File: 167. Two Sum II - Input Array Is Sorted/167.go
// 167. Two Sum II - Input Array Is Sorted solution in GoLang
// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/description/

package main

func twoSum(numbers []int, target int) []int {
	low := 0
	high := len(numbers) - 1

	for low < high {
		sum := numbers[low] + numbers[high]

		if sum == target {
			return []int{low + 1, high + 1}
		} else if sum < target {
			low++
		} else {
			high--
		}
	}

	return []int{-1, -1}
}
